<!DOCTYPE html>
<html lang="zh-CN">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">



<link rel="stylesheet" href="/css/main.css">



<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.2/css/all.min.css">
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/animate.css@3.1.1/animate.min.css">
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.css">
  <script src="https://cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.js"></script>

<script class="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"andriasmatthewi.github.io","root":"/","images":"/images","scheme":"Pisces","version":"8.2.2","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12},"copycode":true,"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"fadeInDown","post_body":"fadeInDown","coll_header":"fadeInLeft","sidebar":"fadeInUp"}},"prism":false,"i18n":{"placeholder":"搜索...","empty":"没有找到任何搜索结果：${query}","hits_time":"找到 ${hits} 个搜索结果（用时 ${time} 毫秒）","hits":"找到 ${hits} 个搜索结果"},"path":"/search.xml","localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false}};
  </script>
<meta name="description" content="GENESPACE使用记录。">
<meta property="og:type" content="article">
<meta property="og:title" content="用GENESPACE建立基于orthologues的染色体共线性关系图">
<meta property="og:url" content="https://andriasmatthewi.github.io/posts/31919/index.html">
<meta property="og:site_name" content="Mirage">
<meta property="og:description" content="GENESPACE使用记录。">
<meta property="og:locale" content="zh_CN">
<meta property="article:published_time" content="2024-04-05T06:00:00.000Z">
<meta property="article:modified_time" content="2024-04-05T14:41:00.835Z">
<meta property="article:author" content="F">
<meta property="article:tag" content="生信">
<meta property="article:tag" content="基因组">
<meta property="article:tag" content="比较基因组学">
<meta name="twitter:card" content="summary">


<link rel="canonical" href="https://andriasmatthewi.github.io/posts/31919/">


<script class="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-CN'
  };
</script>
<title>用GENESPACE建立基于orthologues的染色体共线性关系图 | Mirage</title>
  




  <noscript>
  <style>
  body { margin-top: 2rem; }

  .use-motion .menu-item,
  .use-motion .sidebar,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header {
    visibility: visible;
  }

  .use-motion .header,
  .use-motion .site-brand-container .toggle,
  .use-motion .footer { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle,
  .use-motion .custom-logo-image {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line {
    transform: scaleX(1);
  }

  .search-pop-overlay, .sidebar-nav { display: none; }
  .sidebar-panel { display: block; }
  </style>
</noscript>

<style>.darkmode--activated{--body-bg-color:#282828;--content-bg-color:#333;--card-bg-color:#555;--text-color:#ccc;--blockquote-color:#bbb;--link-color:#ccc;--link-hover-color:#eee;--brand-color:#ddd;--brand-hover-color:#ddd;--table-row-odd-bg-color:#282828;--table-row-hover-bg-color:#363636;--menu-item-bg-color:#555;--btn-default-bg:#222;--btn-default-color:#ccc;--btn-default-border-color:#555;--btn-default-hover-bg:#666;--btn-default-hover-color:#ccc;--btn-default-hover-border-color:#666;--highlight-background:#282b2e;--highlight-foreground:#a9b7c6;--highlight-gutter-background:#34393d;--highlight-gutter-foreground:#9ca9b6}.darkmode--activated img{opacity:.75}.darkmode--activated img:hover{opacity:.9}.darkmode--activated code{color:#69dbdc;background:0 0}button.darkmode-toggle{z-index:9999}.darkmode-ignore,img{display:flex!important}.beian img{display:inline-block!important}</style></head>

<body itemscope itemtype="http://schema.org/WebPage" class="use-motion">
  <div class="headband"></div>

  <main class="main">
    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="切换导航栏" role="button">
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <i class="logo-line"></i>
      <h1 class="site-title">Mirage</h1>
      <i class="logo-line"></i>
    </a>
      <p class="site-subtitle" itemprop="description">There's a million things I haven't done, but just you wait.</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>



<nav class="site-nav">
  <ul class="main-menu menu">
        <li class="menu-item menu-item-home"><a href="/" rel="section"><i class="fa fa-home fa-fw"></i>首页</a></li>
        <li class="menu-item menu-item-about"><a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>关于</a></li>
        <li class="menu-item menu-item-tags"><a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>标签</a></li>
        <li class="menu-item menu-item-categories"><a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>分类</a></li>
        <li class="menu-item menu-item-archives"><a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>归档</a></li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>搜索
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup"><div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off" maxlength="80"
           placeholder="搜索..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close" role="button">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div class="search-result-container no-result">
  <div class="search-result-icon">
    <i class="fa fa-spinner fa-pulse fa-5x"></i>
  </div>
</div>

    </div>
  </div>

</div>
        
  
  <div class="toggle sidebar-toggle" role="button">
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
  </div>

  <aside class="sidebar">

    <div class="sidebar-inner sidebar-nav-active sidebar-toc-active">
      <ul class="sidebar-nav">
        <li class="sidebar-nav-toc">
          文章目录
        </li>
        <li class="sidebar-nav-overview">
          站点概览
        </li>
      </ul>

      <div class="sidebar-panel-container">
        <!--noindex-->
        <div class="post-toc-wrap sidebar-panel">
            <div class="post-toc animated"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#%E4%B8%BA%E4%BB%80%E4%B9%88%E9%80%89%E6%8B%A9GENESPACE"><span class="nav-number">1.</span> <span class="nav-text">为什么选择GENESPACE</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%AE%89%E8%A3%85%E4%B8%8E%E8%BF%90%E8%A1%8C%EF%BC%88%E4%BB%A5%E5%8F%8A%E8%BF%87%E7%A8%8B%E4%B8%AD%E7%9A%84debug%E8%AE%B0%E5%BD%95%EF%BC%89"><span class="nav-number">2.</span> <span class="nav-text">安装与运行（以及过程中的debug记录）</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E8%B0%83%E8%AF%95%E4%B8%8E%E5%88%9D%E5%A7%8B%E5%8C%96"><span class="nav-number">2.1.</span> <span class="nav-text">调试与初始化</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E8%BE%93%E5%85%A5%E6%96%87%E4%BB%B6%E5%87%86%E5%A4%87%EF%BC%88%E6%9C%80%E9%BA%BB%E7%83%A6%E7%9A%84%E4%B8%80%E6%AD%A5%EF%BC%89"><span class="nav-number">2.2.</span> <span class="nav-text">输入文件准备（最麻烦的一步）</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#bed%E6%A0%BC%E5%BC%8F%E7%9A%84%E6%B3%A8%E9%87%8A%E6%96%87%E4%BB%B6"><span class="nav-number">2.2.1.</span> <span class="nav-text">bed格式的注释文件</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#fasta%E6%A0%BC%E5%BC%8F%E7%9A%84pep%E5%A4%9A%E8%82%BD%E5%BA%8F%E5%88%97"><span class="nav-number">2.2.2.</span> <span class="nav-text">fasta格式的pep多肽序列</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E5%B7%A5%E4%BD%9C%E8%B7%AF%E5%BE%84%E4%B8%8E%E8%BE%93%E5%85%A5%E6%96%87%E4%BB%B6%E5%AD%98%E6%94%BE%E4%BD%8D%E7%BD%AE"><span class="nav-number">2.2.3.</span> <span class="nav-text">工作路径与输入文件存放位置</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E8%BF%90%E8%A1%8CGENESPACE-sh"><span class="nav-number">2.3.</span> <span class="nav-text">运行GENESPACE.sh</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E8%B0%83%E7%94%A8%E7%9A%84R%E8%84%9A%E6%9C%ACGENESPACE-R"><span class="nav-number">2.4.</span> <span class="nav-text">调用的R脚本GENESPACE.R</span></a></li></ol></li></ol></div>
        </div>
        <!--/noindex-->

        <div class="site-overview-wrap sidebar-panel">
          <div class="site-author site-overview-item animated" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="F"
      src="/images/avatar.jpg">
  <p class="site-author-name" itemprop="name">F</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap site-overview-item animated">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">21</span>
          <span class="site-state-item-name">日志</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">5</span>
        <span class="site-state-item-name">分类</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">14</span>
        <span class="site-state-item-name">标签</span></a>
      </div>
  </nav>
</div>



        </div>
      </div>
    </div>
  </aside>
  <div class="sidebar-dimmer"></div>


    </header>

    
  <div class="back-to-top" role="button">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>

<noscript>
  <div class="noscript-warning">Theme NexT works best with JavaScript enabled</div>
</noscript>


    <div class="main-inner post posts-expand">


  


<div class="post-block">
  
  

  <article itemscope itemtype="http://schema.org/Article" class="post-content" lang="zh-CN">
    <link itemprop="mainEntityOfPage" href="https://andriasmatthewi.github.io/posts/31919/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.jpg">
      <meta itemprop="name" content="F">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Mirage">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          用GENESPACE建立基于orthologues的染色体共线性关系图
        </h1>

        <div class="post-meta-container">
          <div class="post-meta">
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar"></i>
      </span>
      <span class="post-meta-item-text">发表于</span>
      

      <time title="创建时间：2024-04-05 14:00:00 / 修改时间：22:41:00" itemprop="dateCreated datePublished" datetime="2024-04-05T14:00:00+08:00">2024-04-05</time>
    </span>
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-folder"></i>
      </span>
      <span class="post-meta-item-text">分类于</span>
        <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
          <a href="/categories/%E7%94%9F%E4%BF%A1%E5%88%86%E6%9E%90/" itemprop="url" rel="index"><span itemprop="name">生信分析</span></a>
        </span>
    </span>

  
</div>

            <div class="post-description">GENESPACE使用记录。</div>
        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">
        <h1 id="为什么选择GENESPACE"><a href="#为什么选择GENESPACE" class="headerlink" title="为什么选择GENESPACE"></a>为什么选择GENESPACE</h1><p>目前多数建立染色体共线性分析的软件都是直接基于核酸序列比对结果的，这种分析在用于亲缘关系非常近的物种时是适用的，但如果想要分析属或以上分类关系的物种时往往会因为核算差异太大而得不到足够连续的synteny block，可视化效果非常差。</p>
<p>GENESPACE的分析利用的是orthologues而不是核酸序列本身的比对相似性，因此可以在亲缘关系较远的物种之间找到较为连续的保守区块，从而得到不错的可视化效果。下面是官网教程给出的几种亲缘关系很远的物种染色体共线性分析结果：</p>
<p><a target="_blank" rel="noopener" href="https://htmlpreview.github.io/?https://github.com/jtlovell/tutorials/blob/main/riparianGuide.html">Riparian plotting with GENESPACE (htmlpreview.github.io)</a></p>
<p>&nbsp;</p>
<h1 id="安装与运行（以及过程中的debug记录）"><a href="#安装与运行（以及过程中的debug记录）" class="headerlink" title="安装与运行（以及过程中的debug记录）"></a>安装与运行（以及过程中的debug记录）</h1><p>官网给出的其中一种方法是通过conda单独创建一个环境，将包括orthofinder在内的所有dependency全部装在这里面。但我在安装的时候遇到了很多很多问题，因为官方维护的问题所以在bioconda和conda-forge能够下载到的dependency版本存在大量的不兼容，最后我折腾出了一个折中的方案，即：</p>
<p>①建立一个orthofinder专用的conda环境，；</p>
<p>②建立一个装有R 4.2.3的conda环境，将GENESPACE这个R包需要的其他R包装在这个环境中；</p>
<p>③在实际使用的时候激活orthofinder的环境，并直接在那里跨环境调用R 4.2.3；</p>
<p>④在conda以外手动安装MCScanX。</p>
<p>感觉相当于曲线救国了，整个安装调试下来差不多折腾了两个月。不过这样做的好处是不会轻易搞乱早已调试好的环境里的dependency，以防之后出错。</p>
<h2 id="调试与初始化"><a href="#调试与初始化" class="headerlink" title="调试与初始化"></a>调试与初始化</h2><p>在R环境下：</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">library(GENESPACE)</span><br><span class="line"></span><br><span class="line">wd &lt;- <span class="string">&quot;/GENESPACE/workingDirectory&quot;</span></span><br><span class="line">path2mcscanx &lt;- <span class="string">&quot;/home/user/software/MCScanX/MCScanX-master&quot;</span></span><br><span class="line">gpar &lt;- init_genespace(wd = wd, path2mcscanx = path2mcscanx, nCores = <span class="number">10</span>)</span><br></pre></td></tr></table></figure>

<p>此时得到了如下的报错：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">Checking dependencies ...</span><br><span class="line">/home/user/miniconda3/envs/orthofinder/bin/scripts_of/tree.py:367: SyntaxWarning: invalid escape sequence &#x27;\-&#x27;</span><br><span class="line">  &quot;&quot;&quot;</span><br><span class="line">/home/user/miniconda3/envs/orthofinder/bin/scripts_of/tree.py:1422: SyntaxWarning: invalid escape sequence &#x27;\-&#x27;</span><br><span class="line">  &quot;&quot;&quot;</span><br><span class="line">/home/user/miniconda3/envs/orthofinder/bin/scripts_of/newick.py:54: SyntaxWarning: invalid escape sequence &#x27;\[&#x27;</span><br><span class="line">  _ILEGAL_NEWICK_CHARS = &quot;:;(),\[\]\t\n\r=&quot;</span><br><span class="line">/home/user/miniconda3/envs/orthofinder/bin/scripts_of/newick.py:57: SyntaxWarning: invalid escape sequence &#x27;\[&#x27;</span><br><span class="line">  _NHX_RE = &quot;\[&amp;&amp;NHX:[^\]]*\]&quot;</span><br><span class="line">/home/user/miniconda3/envs/orthofinder/bin/scripts_of/newick.py:58: SyntaxWarning: invalid escape sequence &#x27;\d&#x27;</span><br><span class="line">  _FLOAT_RE = &quot;[+-]?\d+\.?\d*(?:[eE][-+]\d+)?&quot;</span><br><span class="line">/home/user/miniconda3/envs/orthofinder/bin/scripts_of/newick.py:60: SyntaxWarning: invalid escape sequence &#x27;\[&#x27;</span><br><span class="line">  _NAME_RE = &quot;[^():,;\[\]]+&quot;</span><br><span class="line">/home/user/miniconda3/envs/orthofinder/bin/scripts_of/newick.py:337: SyntaxWarning: invalid escape sequence &#x27;\s&#x27;</span><br><span class="line">  MATCH = &#x27;%s\s*%s\s*(%s)?&#x27; % (FIRST_MATCH, SECOND_MATCH, _NHX_RE)</span><br><span class="line">/home/user/miniconda3/envs/orthofinder/bin/scripts_of/probroot.py:10: SyntaxWarning: invalid escape sequence &#x27;\i&#x27;</span><br><span class="line">  &quot;&quot;&quot;</span><br><span class="line">/home/user/miniconda3/envs/orthofinder/bin/scripts_of/probroot.py:201: SyntaxWarning: invalid escape sequence &#x27;\l&#x27;</span><br><span class="line">  &quot;&quot;&quot;</span><br><span class="line">/home/user/miniconda3/envs/orthofinder/bin/scripts_of/probroot.py:267: SyntaxWarning: invalid escape sequence &#x27;\l&#x27;</span><br><span class="line">  &quot;&quot;&quot;</span><br><span class="line">        Found valid path to OrthoFinder v2.55: `orthofinder`</span><br><span class="line">        Found valid path to DIAMOND2 v2.19: `diamond`</span><br><span class="line">        Found valid MCScanX_h executable: `/home/user/software/MCScanX/MCScanX-master/MCScanX_h`</span><br></pre></td></tr></table></figure>

<p>个人感觉这里不是软件本身的问题，而是因为我在安装的时候跨了太多个环境，中途更换的dependency版本可能导致python版本的不兼容，在调用tree.py、newick.py和probroot.py这三个脚本时检测的raw strings字符串包含了“\”这种正则符号，因此无法正常运行。最后的解决方式就是挨个更改这三个python脚本，在出现报错那几行的“”符号前加上一个“r”，代表这后面引号框起来的都是纯字符串而不是正则符，以防被系统检测为syntax error。具体例子：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">vim /home/user/miniconda3/envs/orthofinder/bin/scripts_of/newick.py</span><br><span class="line"></span><br><span class="line">#原先报错的第54行是：</span><br><span class="line">_ILEGAL_NEWICK_CHARS = &quot;:;(),\[\]\t\n\r=&quot;</span><br><span class="line"></span><br><span class="line">#把它更改成：</span><br><span class="line">_ILEGAL_NEWICK_CHARS = r&quot;:;(),\[\]\t\n\r=&quot;</span><br></pre></td></tr></table></figure>

<p>这样就可以解决报错问题了。</p>
<h2 id="输入文件准备（最麻烦的一步）"><a href="#输入文件准备（最麻烦的一步）" class="headerlink" title="输入文件准备（最麻烦的一步）"></a>输入文件准备（最麻烦的一步）</h2><p>GENESPACE本身是有一个<code>parse_annotations()</code>的功能的。但是软件预设能够检测的注释格式很少，在教程里只有“ncbi”和“phytozome”两个格式。我用到的几个基因组都是从ensembl下载的，注释格式和GENESPACE能检测的不一致，所以需要自己手动转换格式。在转换过程中要格外注意检查每一步的中间文件，确保得到的文件能被GENESPACE正确识别。</p>
<h3 id="bed格式的注释文件"><a href="#bed格式的注释文件" class="headerlink" title="bed格式的注释文件"></a>bed格式的注释文件</h3><p>GENESPACE本身需要的bed文件格式如下：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">chr1  1500  2000  gene1</span><br><span class="line">chr1  2500  3200  gene2</span><br></pre></td></tr></table></figure>

<p>而我本身的注释文件是ensembl的gff3格式，因此这里要用BEDOPS软件包中的gff2bed工具转换格式，并去除分类为“gene”以外的其他类别，比如“mRNA”“CDS”和“exon”之类的。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"># Generating bed file from gff3 annotations</span><br><span class="line">gff2bed &lt; species.gff3 \</span><br><span class="line">&gt; species_raw.bed</span><br><span class="line"></span><br><span class="line"># Reformatting bed files</span><br><span class="line">awk -F &quot;\t&quot; &#x27;$8 == &quot;gene&quot; &#123;print $1, $2, $3, $4&#125;&#x27; species_raw.bed \</span><br><span class="line">&gt; species.bed</span><br></pre></td></tr></table></figure>

<h3 id="fasta格式的pep多肽序列"><a href="#fasta格式的pep多肽序列" class="headerlink" title="fasta格式的pep多肽序列"></a>fasta格式的pep多肽序列</h3><p>这个文件需要从下载参考基因组的地方找，有些基因组组装完成度没达到这个级别所以很可能只有核酸序列没有蛋白组序列，为了保持基因组质量的一致性，我在这里选的基因组都是本身就自带pep.fa的那些。</p>
<p>fasta文件有两个需要手动更正的格式问题，第一是header太长太冗余，我们需要的只有“gene:”后面接的那个编号而已；第二是在序列中包含了大量的换行，需要把它们全都删掉，变成如下的格式：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">&gt;gene1</span><br><span class="line">MGASGRGAGEQQSPSSTGMLVMSECKGRDRSPSSSMMLVMSECKGRDRSPSSSMMLVMSECKGRDRSPSSSMMLVMSECKGRDRSPSSSMMLVMSECKGRDRSPSSSMMLVMSECKGRDRSPSSSM</span><br><span class="line">&gt;gene2</span><br><span class="line">MLVMSECKGRDRSPSSSMRDRSPSSSMMLVMSECKGRDRSPSSRDRSPSSSMMLVMSECKGRDRSPSSRDRSPSSSMMLVMSECKGRDRSPSSRDRSPSSSMMLVMSECKGRDRSPSSRDRSPSSSMMLVMSECKGRDRSPSS</span><br></pre></td></tr></table></figure>

<p>首先用seqtk软件去除fasta中的换行，之后再用awk对每个蛋白序列的header进行格式转换：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"># Removing multiple lines in original fasta</span><br><span class="line">seqtk seq -l 0 species.pep.fa \</span><br><span class="line">&gt; species.pep_raw.fa</span><br><span class="line"></span><br><span class="line"># Parsing fasta headers</span><br><span class="line">awk &#x27;$0 ~ &quot;^&gt;&quot; &#123;match($0, /gene:([0-9A-Z_\.]+)/, gene); print &quot;&gt;&quot;gene[1]; next;&#125;1&#x27; species.pep_raw.fa \</span><br><span class="line">&gt; species.fa</span><br></pre></td></tr></table></figure>

<p>▲这里要注意一点，每个物种编辑好的bed和fa文件命名必须是相同的，比如species.bed对应的就只能是species.fa，否则GENESPACE也会检测不到。</p>
<h3 id="工作路径与输入文件存放位置"><a href="#工作路径与输入文件存放位置" class="headerlink" title="工作路径与输入文件存放位置"></a>工作路径与输入文件存放位置</h3><p>建立一个空文件夹（比如叫“/GENESPACE/workingDirectory”），将上一步编辑好的几个bed和fa文件分别移动至里面相应的位置：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">/workingDirectory</span><br><span class="line">└─ peptide</span><br><span class="line">    └─ species1.fa</span><br><span class="line">    └─ species2.fa</span><br><span class="line">└─ bed</span><br><span class="line">    └─ species1.bed</span><br><span class="line">    └─ species2.bed</span><br></pre></td></tr></table></figure>

<h2 id="运行GENESPACE-sh"><a href="#运行GENESPACE-sh" class="headerlink" title="运行GENESPACE.sh"></a>运行GENESPACE.sh</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">source activate orthofinder</span><br><span class="line"></span><br><span class="line">/home/user/miniconda3/envs/R_base_4/lib/R/bin/R --slave --vanilla &lt; GENESPACE.R &gt; GENESPACE.Rout</span><br></pre></td></tr></table></figure>

<h2 id="调用的R脚本GENESPACE-R"><a href="#调用的R脚本GENESPACE-R" class="headerlink" title="调用的R脚本GENESPACE.R"></a>调用的R脚本GENESPACE.R</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">library(GENESPACE)</span><br><span class="line"></span><br><span class="line">wd &lt;- <span class="string">&quot;/GENESPACE/workingDirectory&quot;</span></span><br><span class="line">path2mcscanx &lt;- <span class="string">&quot;/home/user/software/MCScanX/MCScanX-master&quot;</span></span><br><span class="line">gpar &lt;- init_genespace(wd = wd, path2mcscanx = path2mcscanx, nCores = <span class="number">10</span>)</span><br><span class="line"></span><br><span class="line">out &lt;- run_genespace(gpar)</span><br></pre></td></tr></table></figure>
    </div>

    
    
    

    <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/%E7%94%9F%E4%BF%A1/" rel="tag"># 生信</a>
              <a href="/tags/%E5%9F%BA%E5%9B%A0%E7%BB%84/" rel="tag"># 基因组</a>
              <a href="/tags/%E6%AF%94%E8%BE%83%E5%9F%BA%E5%9B%A0%E7%BB%84%E5%AD%A6/" rel="tag"># 比较基因组学</a>
          </div>

        

          <div class="post-nav">
            <div class="post-nav-item">
                <a href="/posts/64077/" rel="prev" title="用BRAKER3对无注释的参考基因组进行基因预测">
                  <i class="fa fa-chevron-left"></i> 用BRAKER3对无注释的参考基因组进行基因预测
                </a>
            </div>
            <div class="post-nav-item">
            </div>
          </div>
    </footer>
  </article>
</div>






    <div class="comments" id="comments">
      <script src="https://utteranc.es/client.js"
              repo="andriasmatthewi/Blog_comment"
              issue-term="pathname"
              theme="preferred-color-scheme"
              crossorigin="anonymous"
              async>
      </script>
    </div>

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      const activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      const commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

</div>
  </main>

  <footer class="footer">
    <div class="footer-inner">


<div class="copyright">
  &copy; 
  <span itemprop="copyrightYear">2024</span>
  <span class="with-love">
    <i class="fas fa-atom"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">F</span>
</div>

<!--
  <div class="powered-by">由 <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://theme-next.js.org/pisces/" class="theme-link" rel="noopener" target="_blank">NexT.Pisces</a> 强力驱动
  </div>-->

    </div>
  </footer>

  
  <script src="https://cdn.jsdelivr.net/npm/animejs@3.2.1/lib/anime.min.js"></script>
<script src="/js/utils.js"></script><script src="/js/motion.js"></script><script src="/js/next-boot.js"></script>

  
<script src="/js/local-search.js"></script>





  <script>
    NProgress.configure({
      showSpinner: true
    });
    NProgress.start();
    document.addEventListener('readystatechange', () => {
      if (document.readyState === 'interactive') {
        NProgress.inc(0.8);
      }
      if (document.readyState === 'complete') {
        NProgress.done();
      }
    });
    document.addEventListener('pjax:send', () => {
      NProgress.start();
    });
    document.addEventListener('pjax:success', () => {
      NProgress.done();
    });
  </script>


  




<script src="https://unpkg.com/darkmode-js@1.5.7/lib/darkmode-js.min.js"></script>

<script>
var options = {
  bottom: '64px',
  right: '32px',
  left: 'unset',
  time: '0.5s',
  mixColor: 'transparent',
  backgroundColor: 'transparent',
  buttonColorDark: '#100f2c',
  buttonColorLight: '#fff',
  saveInCookies: true,
  label: '🌓',
  autoMatchOsTheme: true
}
const darkmode = new Darkmode(options);
window.darkmode = darkmode;
darkmode.showWidget();
</script>

<script>
NexT.utils.loadComments('.utterances-container', () => {
  const script = document.createElement('script');
  script.src = 'https://utteranc.es/client.js';
  script.setAttribute('repo', "andriasmatthewi/Blog_comment");
  script.setAttribute('issue-term', "pathname");
  script.setAttribute('theme', "preferred-color-scheme");
  script.crossOrigin = 'anonymous';
  script.async = true;
  document.querySelector('.utterances-container').appendChild(script);
});
</script>

</body>
</html>
